lib/smooth/application/io/i2c/BME280.cpp

```diff
@@ -59,217 +61,118 @@ namespace smooth::application::sensor
-                && read_16bit(DIG_P8_REG, trimming.dig_P8)
-                && read_16bit(DIG_P9_REG, trimming.dig_P9)
-                && read_8bit(DIG_H1_REG, trimming.dig_H1)
-                && read_16bit(DIG_H2_REG, trimming.dig_H2)
-                && read_8bit(DIG_H3_REG, trimming.dig_H3)
-                && read_8bit(DIG_H6_REG, trimming.dig_H6)
-
-                // The following data is a bit tricky as it is split over multiple bytes.
-                // dig_H4: Bits 11:4 located in DIG_H4_REG, bits 3:0 located in DIG_H4_REG+1
-                // dig_H5: bits 3:0 located in DIG_H5_REG[7:4], bits 11:4 in DIG_H5_REG+1
-
-                && read_8bit(DIG_H4_REG, h4)
-                && read_8bit(static_cast<uint8_t>(DIG_H4_REG + 1), h4_1)
-                && read_8bit(DIG_H5_REG, h5)
-                && read_8bit(static_cast<uint8_t>(DIG_H5_REG + 1), h5_1);
-
-            // Adjust values so they are directly usable.
-            trimming.dig_H4 = static_cast<int16_t>((h4 << 4) | (h4_1 & 0xF));
-            trimming.dig_H5 = static_cast<int16_t>((h5 >> 4) | (h5_1 << 4));
+            core::util::FixedBuffer<uint8_t, 32> calibration_data;
+            core::util::FixedBuffer<uint8_t, 26> calib00_calib25_data;    // 0x88-0xA1
+            core::util::FixedBuffer<uint8_t, 7> calib26_calib32_data;     // 0xE1-0xE7
+
+            trimming_read = read(address, BME280Core::CALIB00_REG, calib00_calib25_data) &&
+                            read(address, BME280Core::CALIB26_REG, calib26_calib32_data);
+
+            if (trimming_read)
+            {
+                // copy 0x88-0x9F into calibration_data
+                for (size_t i = 0; i < 24; i++)
+                {
+                    calibration_data[i] = calib00_calib25_data[i];
+                }
+
+                // copy 0xA1 but not 0xA0 into calibration_data
+                calibration_data[24] = calib00_calib25_data[25];
+
+                // copy 0xE1 to 0xE7 into calibration_data
+                for (size_t i = 0; i < 7; i++)
+                {
+                    calibration_data[i + 25] = calib26_calib32_data[i];
+                }
+
+                bme280_core.populate_trimming_registers(calibration_data);
+            }
+            else
+            {
+                Log::error(TAG, "Problems reading trimming parameters");
+            }
         }
 
         return trimming_read;
     }
 
+    // Read Temperature, Pressure and Humidity
     bool BME280::read_measurements(float& humidity, float& pressure, float& temperature)
     {
-        core::util::FixedBuffer<uint8_t, 8> measurement{};
+        core::util::FixedBuffer<uint8_t, 8> measurement_data{};
 
         humidity = 0;
         pressure = 0;
         temperature = 0;
 
         bool res = read_trimming_parameters();
-        res = res && read(address, PRESS_MSB_REG, measurement);
+        res = res && read(address, BME280Core::PRESS_MSB_REG, measurement_data);
```

Comment:
Seeing as these two assignments to res are beside eachother I think we can simplify this to just 
```
res = read(...) 
    && read(..)
```

Reply:
Updated